src/BenchmarkDotNet/Loggers/SynchronousProcessOutputLoggerWithDiagnoser.cs

```diff
@@ -38,22 +38,30 @@ public SynchronousProcessOutputLoggerWithDiagnoser(ILogger logger, Process proce
         internal void ProcessInput()
         {
             string line;
-            while ((line = process.StandardOutput.ReadLine()) != null)
+
+            // ReadLine() can return string.Empty/null as values.
+            while (!process.StandardOutput.EndOfStream)
             {
-                logger.WriteLine(LogKind.Default, line);
+                if (process.StandardOutput.Peek() > 0) // 0 or -1 can indicate internal error.
+                {
+                    line = process.StandardOutput.ReadLine();
 
-                if (string.IsNullOrEmpty(line))
-                    continue;
+                    if (!string.IsNullOrEmpty(line)) // Skip bad data.
+                    {
+                        logger.WriteLine(LogKind.Default, line);
 
-                if (!line.StartsWith("//"))
-                    LinesWithResults.Add(line);
-                else if (Engine.Signals.TryGetSignal(line, out var signal))
-                {
-                    diagnoser?.Handle(signal, diagnoserActionParameters);
-                    process.StandardInput.WriteLine(Engine.Signals.Acknowledgment);
+                        if (!line.StartsWith("//"))
+                        { LinesWithResults.Add(line); }
+                        else if (Engine.Signals.TryGetSignal(line, out var signal))
+                        {
+                            diagnoser?.Handle(signal, diagnoserActionParameters);
+                            process.StandardInput.WriteLine(Engine.Signals.Acknowledgment);
+                        }
+                        else
+                        { LinesWithExtraOutput.Add(line); }
+                    }
                 }
-                else
-                    LinesWithExtraOutput.Add(line);
+                else { break; } // No more 'characters' can be read.
```

Comment:
could you please remove the `else { break; }`  and move the condition to while loop?

`while (!process.StandardOutput.EndOfStream && process.StandardOutput.Peek() > 0)`